kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveRestoreConsumer.java

```diff
@@ -1,40 +1,65 @@
 package dev.responsive.kafka.clients;
 
+import dev.responsive.kafka.store.ResponsiveRestoreListener;
+import java.time.Duration;
 import java.util.Collection;
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Objects;
 import java.util.OptionalLong;
+import java.util.Set;
 import java.util.function.Function;
 import org.apache.kafka.clients.consumer.Consumer;
 import org.apache.kafka.clients.consumer.OffsetAndMetadata;
 import org.apache.kafka.common.TopicPartition;
 
 /**
  * Consumer implementation to use for restoring Responsive stores. This Restore Consumer emulates
- * consumping a topic that has been truncated up to some offset on a subset of topic partitions.
+ * consuming a topic that has been truncated up to some offset on a subset of topic partitions.
  * The specific offsets are provided by a function mapping the topic partition to the start
  * offset. This consumer is used in conjunction with Responsive Stores, which provide a safe
  * starting offset that has been committed to a remote storage system. This is useful for
  * minimizing the length of restores when using a Responsive Store over a compacted changelog
  * topic (which cannot be truncated).
+ * <p>
+ * This consumer is used by Kafka Streams for both active and standby task restoration, though
+ * only one or the other at a given time. This may be subject to change, but the general process
+ * follows this pattern:
+ * 1. initialize changelogs for newly assigned tasks:
+ *     i. restoreConsumer.assign(current assignment + new changelogs)
+ *     ii. if STANDBY_MODE: restoreConsumer.pause(standby changelogs)
+ *     iii. restoreConsumer.seekToBeginning(new changelogs)
+ *     iv. startOffset = restoreConsumer.position(changelog)
+ *     iv. for each ACTIVE changelog: call StateRestoreListener#onRestoreStart
+ * 2. restore active/standby state stores:
+ *     i. poll for new record batch
+ *     ii. restore batch through StateRestoreCallback
+ *     iii. for each ACTIVE changelog in batch: call StateRestoreListener#onBatchRestored
+ *     iv. update current offset based on highest offset in batch
  *
  * @param <K> Type of the record keys
  * @param <V> Type of the record values
  */
 public class ResponsiveRestoreConsumer<K, V> extends DelegatingConsumer<K, V> {
```

Comment:
I'm not sure how confident you are about the assumptions around what state the task is in (active vs standby) at different points. But we could wire that information in via the store and assert on it here.